src/core/public/chrome/chrome_service.ts

```diff
@@ -0,0 +1,145 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import * as Url from 'url';
+
+import * as Rx from 'rxjs';
+import { map, takeUntil } from 'rxjs/operators';
+
+const IS_COLLAPSED_KEY = 'core.chrome.isCollapsed';
+
+function isEmbedParamInHash() {
+  const { query } = Url.parse(String(window.location.hash).slice(1), true);
+  return Boolean(query.embed);
+}
+
+export interface Brand {
+  logo?: string;
+  smallLogo?: string;
+}
+
+export class ChromeService {
+  private readonly stop$ = new Rx.ReplaySubject(1);
+
+  public start() {
+    const FORCE_HIDDEN = isEmbedParamInHash();
+
+    const brand$ = new Rx.BehaviorSubject<Brand>({});
+    const isVisible$ = new Rx.BehaviorSubject(true);
+    const isCollapsed$ = new Rx.BehaviorSubject(!!localStorage.getItem(IS_COLLAPSED_KEY));
+    const applicationClasses$ = new Rx.BehaviorSubject<Set<string>>(new Set());
+
+    return {
+      /**
+       * Set the brand configuration. Normally the `logo` property will be rendered as the
+       * CSS background for the home link in the chrome navigation, but when the page is renderd
+       * in a small window the `smallLogo` will be used and rendered at about 45px wide.
+       *
+       * example:
+       *
+       *    chrome.setBrand({
+       *      logo: 'url(/plugins/app/logo.png) center no-repeat'
+       *      smallLogo: 'url(/plugins/app/logo-small.png) center no-repeat'
+       *    })
+       *
+       */
+      setBrand: (brand: Brand) => {
+        brand$.next(
+          Object.freeze({
+            logo: brand.logo,
+            smallLogo: brand.smallLogo,
+          })
+        );
+      },
+
+      /**
+       * Get an observable of the current brand information.
+       */
+      getBrand$: () => brand$.pipe(takeUntil(this.stop$)),
+
+      /**
+       * Set the temporary visibility for the chrome. This does nothing if the chrome is hidden
+       * by default and should be used to hide the chrome for things like full-screen modes
+       * with an exit button.
+       */
+      setIsVisible: (visibility: boolean) => {
+        isVisible$.next(visibility);
+      },
+
+      /**
+       * Get an observable of the current visiblity state of the chrome.
+       */
+      getIsVisible$: () =>
```

Comment:
I think something like `FORCE_HIDDEN ? of(false) : ...` would be fine, but I'm kind of concerned about the difference in lifecycle between the two return values, it probably wouldn't be a big deal, and since it really can't change it is "complete", but I'd have to think through how that might impact consumers. As for deduping here rather than in consumers, I originally had more dedupe logic in here, but felt like it didn't really add anything to do the extra work of deduping notifications. It might turn out that it's a lot nicer to consume an API that properly dedupes notifications, but I'm not sure about that... So I left it out for now.